src/classes/ADDR_CopyAddrHHObjBTN_TEST.cls

```diff
@@ -200,6 +200,55 @@ private with sharing class ADDR_CopyAddrHHObjBTN_TEST {
         System.assert(mockPerms.updateSObjFields.contains(npo02__Household__c.npo02__MailingStreet__c));
     }
 
+    /***
+    * @description Verifies an error message is displayed when a user does not have household update permissions
+    * @return void
+    */
+    @isTest
+    private static void shouldErrorWhenBtnClickedWithoutCreateAddressPerms() {
+        List<Address__c> addExists = new List<Address__c>();
+
+        getRecordData();
+        mockPerms.hasCreate = false;
+        createCopyAddressBtnCtrl(householdRecord.Id);
+
+        copyAddressCtrl.buttonClick();
+        addExists = getAddressRecord();
+
+        UTIL_UnitTestData_TEST.assertPageHasError(System.Label.addrCopyConAddBtnFls);
+        System.assertEquals(0, addExists.size(), 'Address record should not be created with Household address');
+        System.assertEquals(Address__c.SObjectType, mockPerms.createSObjType, 'The object type should match');
+        System.assert(mockPerms.createSObjFields.contains(Address__c.MailingStreet__c));
+    }
+
+    /***
+    * @description Verifies if address record is created when user has necessary permissions
+    * @return void
+    */
+    @isTest
+    private static void shouldCreateOrCopyAddressWhenBtnClickedWithCreateAddressPerms() {
```

Comment:
- The ApexDoc need to be updated to indicate what permissions are required. There is no need for `@return` in the test.
- Has the functionality for this test already tested in existing unit tests such as `shouldCopyAddressFromContact()` and `shouldCopyAddressFromHousehold()`?

Reply:
@RadSFDeveloper , I know those tests verify if address has been copied from household to contact and vice versa. However, those tests don't verify if the address__c record has been created and set as default or not.